xdsl/dialects/vector.py

```diff
@@ -383,6 +397,330 @@ def __init__(
+
+            minor_size = (
+                1 if vector_type.get_num_dims() == 0 else vector_type.get_shape()[-1]
+            )
+            result_vec_size = vector_element_type.bitwidth * minor_size
+            if result_vec_size % element_type.bitwidth != 0:
+                raise VerifyException(
+                    f'"{op.name}" requires the bitwidth of the minor 1-D vector to be an integral multiple of the bitwidth of the source element type'
+                )
+
+        # Check that permutation map results match rank of vector type.
+        if len(permutation_map.results) != vector_type.get_num_dims():
+            raise VerifyException(
+                f'"{op.name}" requires a permutation_map with result dims of the same rank as the vector type'
+            )
+
+    if permutation_map.num_symbols != 0:
+        raise VerifyException(f'"{op.name}" requires permutation_map without symbols')
+
+    if permutation_map.num_dims != shaped_type.get_num_dims():
+        raise VerifyException(
+            f'"{op.name}" requires a permutation_map with input dims of the same rank as the source type'
+        )
+
+    # WJOG9GVF: TODO fix: uncomment this when 7S4F0FZA has been fixed
+    # if mask_type:
+    #     if mask_type != inferred_mask_type:
+    #         raise VerifyException(
+    #             f'"{op.name}" inferred mask type ({inferred_mask_type}) and mask operand type ({mask_type}) don\'t match'
+    #         )
+
+    if len(in_bounds) != len(permutation_map.results):
+        raise VerifyException(
+            f'"{op.name}" expects the optional in_bounds attr of same rank as permutation_map results: {str(permutation_map)} vs in_bounds of of size {len(in_bounds)}'
+        )
+
+    for i in range(len(permutation_map.results)):
+        if (
+            isa(permutation_map.results[i], AffineConstantExpr)
+            and not in_bounds.data[i].value.data
+        ):
+            raise VerifyException(
+                f'"{op.name}" requires broadcast dimensions to be in-bounds'
+            )
+
+
+def infer_transfer_op_mask_type(
+    vector_type: VectorType[Attribute],
+    affine_map: AffineMap,
+) -> VectorType[I1] | None:
+    """
+    TODO test
+    """
+
+    # 7S4F0FZA
+    # TODO uncomment and test this once VectorType has been fixed, see issue #3654
+    # When you do this also fix all WJOG9GVF
+
+    # inverse_permutation_map = affine_map.compress_dims(
+    #     affine_map.unused_dims_bit_vector()
+    # ).inverse_permutation()
+
+    # assert inverse_permutation_map
+
+    # mask_shape = inverse_permutation_map.compose_with_values(vector_type.get_shape())
+
+    # scalable_dims = inverse_permutation_map.eval(
+    #     [1 if dim_scalable else 0 for dim_scalable in vector_type.get_scalable_dims()],
+    #     [],
+    # )
+
+    # return VectorType(
+    #     i1,
+    #     mask_shape,
+    #     [dim_scalable == 1 for dim_scalable in scalable_dims],
+    # )
```

Comment:
?

Reply:
The problem is issue #3654 
As soon as this is fixed the code above can be  uncommented. I can remove it but then how should I document this?

Reply:
I can move it to a new pull request and link it to the issue